agent/agent_command_test.go

```diff
@@ -57,6 +57,28 @@ func (s *CommandSuite) SetupTest() {
 	}
 }
 
+func (s *CommandSuite) TestPreErrorFailsWithSetup() {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	taskID := "pre_error"
+	s.tc.task.ID = taskID
+	s.tc.ranSetupGroup = false
+
+	defer s.a.removeTaskDirectory(s.tc)
+	_, err := s.a.runTask(ctx, s.tc)
+	s.NoError(err)
+	detail := s.mockCommunicator.GetEndTaskDetail()
+	s.Equal(evergreen.TaskFailed, detail.Status)
+	s.Equal(evergreen.CommandTypeSetup, detail.Type)
```

Comment:
If `pre_error_fails_task` is true, should this respect the command/top-level `command_type`? I would have thought based on the YAML `command_type: system` that it would result in a system failure instead of setup failure.

Reply:
The top-level command gets [overridden](https://github.com/evergreen-ci/evergreen/blob/main/model/project.go#L1180) by the current command's type, so since the failing pre command specifies `setup` as its type it will be the type that is respected.